Codecademy/gamut
# refactor(Badge): updated Badge variants' color names

### Overview
Addressing [WEB-2446](https://codecademy.atlassian.net/browse/WEB-2446) -- 
Adjust dark-mode`grey` badges variants to meet WCAG contrast. 

Expanded the scope to update all variants names, see figma attached: 

### PR Checklist

- [x] Related to designs: https://www.figma.com/file/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?type=design&node-id=29959-39721&mode=design&t=InPLnttaWm84srkv-0 
- [x] Related to JIRA ticket: [WEB-2446]

#### Testing Instructions

Load up storybook and find the badges page: http://localhost:6006/?path=/docs/atoms-badge--badge 
The badges should still load just fine. 



#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/36634)  | [Monolith Env](https://pr-36634-monolith.dev-eks.codecademy.com/catalog?PR_ENV=pr-36634-portal) |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/3934)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-3934)   |




[WEB-2446]: https://codecademy.atlassian.net/browse/WEB-2446?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[WEB-2446]: https://codecademy.atlassian.net/browse/WEB-2446?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
